### PR TITLE
Test

### PR DESCRIPTION
**Pull request by tarikbouari**
In this milestone we implemented the following tasks: 
-  Setup React Testing Library.
- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components
- Use Jest snapshots to test the components.
- Use React Testing Library to simulate user interaction.
- Refactor code and fix linters errors

### CODE REVIEW

File: src/logic/Calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+describe('calculate', () => {
+  test('should return an object', () => {
+    const data = {
+      total: '21',
+      next: '7',
+      operation: null,
+    };
+    const result = { total: '7', next: null, operation: '/' };
+    expect(calculate(data, '/')).toEqual(result);
+  });
```
Comment (afanimartin) [inline src/logic/Calculate.test.js:12]: - Please add egde cases for `buttonName === AC`, `buttonName === '.'`, `buttonName == '='` and `buttonName == '+/-'`. All functions and scenarios must be fully tested.

File: src/logic/Operate.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  expect(operate(10, 2, '+')).toBe('12');
+  expect(operate(10, 2, '-')).toBe('8');
+  expect(operate(10, 2, 'x')).toBe('20');
+  expect(operate(10, 2, '÷')).toBe('5');
+  expect(operate(11, 2, '%')).toBe('1');
+  expect(operate(1, 2, '÷')).toBe('0.5');
+  expect(operate(10, 0, '÷')).toBe("Can't divide by 0.");
+  expect(operate(10, -2, '÷')).toBe('-5');
+  expect(operate(10, 0, '%')).toBe("Can't find modulo as can't divide by 0.");
+});
```
Comment (afanimartin) [inline src/logic/Operate.test.js:13]: - Add more tests to cover negative edge cases as well

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @tarikbouari, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🎉 🎉 🎉 
- Tests added as required
- Readme looks professional

## Required Changes ♻️
_Check the comments under the review._
- Please fix your tests. All tests are failing.
Screenshot from 2022-09-23 12-56-50


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — COMMENTED** [review]
Hi team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

I think you submitted the project in the wrong place. I can see the requirements for creating a new react project and I am seeing that you have created snapshots for tests. I already started the review so I decided to leave the optional suggestions I already made so that you can check them out before. your proper submission.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Optional Suggestions
[OPTIONAL] Kindly include information on how to use your tests in the README
[OPTIONAL] Kindly remove the `if available ` text in your README beside the `Live Demo`  text.
[OPTIONAL] GOD JOB 👍🏽 You all wrote your commit messages in the imperative. But in the feature kindly remember that you should also follow. the rules of capitalization in the English language. Some of you started the commit message in lowercase.
[OPTIONAL] Your `Live Demo` link in your README is currently not working.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by ahmetbozaci — APPROVED** [review]
# 🥳 🎉🥇 💯  💥 🎆 🎈 STATUS: APPROVED 🥳 🎉  🥇  💯  💥   🎆 🎈 🎉 



@ ✋🏼 
#### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.
___ 

### 💯 Highlights 💯
- You added tests and all tests passed. ✔️ 
image


___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ 

- Your commit messages should always start with capital letters and they should be an imperative tense. I see some of them are not like that. Instead of writing `Added style or Created the file` kindly write `Add style or Create the file`
- ***I write it as optional. But this is a [general requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=The%20commit%20message%20should%20be%20written%20in%20the%20imperative%20tense%20and%20capitalized) for every project. So you should pay attention to this in your next projects.***
- Kindly change them by using `git-rebase`. If you want to know how to change ı leave a few resources.
- [Video - How to change commit messages](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6)
- [Article - How to create good commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)

⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ 
___

### Cheers and Happy coding! 💻 👏
aproved


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_
